database/sql/schema.sql

```diff
@@ -39,8 +39,8 @@ CREATE TABLE HistoricalData(
 
 CREATE TABLE CourseOffering (
 	id BINARY(16) PRIMARY KEY,
-	course_name VARCHAR(100) NOT NULL,
-	course_code VARCHAR(10) NOT NULL, 
+	course_name VARCHAR(100) NOT NULL UNIQUE,
+	course_code VARCHAR(10) NOT NULL UNIQUE, 
 	course_desc VARCHAR(2000) NOT NULL,
 	prof_prereq VARCHAR(500),
 	min_offering INT NOT NULL,
```

Comment:
We now support saving historical preferences for profs therefore we need to add a unique constraint in ProfessorAvailability table: 

- [ ] Add `UNIQUE KEY `my_uniq_id` (`year`,`prof_id`)` to ProfessorAvilability table